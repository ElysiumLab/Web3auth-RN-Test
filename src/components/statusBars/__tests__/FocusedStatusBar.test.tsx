import React from "react";
import { render } from "@testing-library/react-native";

import FocusedStatusBar from "../FocusedStatusBar";

jest.mock("@react-navigation/core", () => {
  const actualNav = jest.requireActual("@react-navigation/core");
  return {
    ...actualNav,
    useIsFocused: jest
      .fn()
      .mockReturnValueOnce(true)
      .mockReturnValueOnce(false),
  };
});

describe("<FocusedStatusBar />", () => {
  it("renders correctly", () => {
    const tree = render(<FocusedStatusBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("unfocused correctly", () => {
    const tree = render(<FocusedStatusBar />).toJSON();
    expect(tree).toBeNull();
  });
});
