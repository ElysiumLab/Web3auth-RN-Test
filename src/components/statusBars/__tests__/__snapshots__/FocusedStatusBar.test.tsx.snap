// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FocusedStatusBar /> renders correctly 1`] = `null`;
